matchms/filtering/metadata_processing/derive_annotation_from_compound_name.py

```diff
@@ -40,13 +40,13 @@ def derive_smiles_from_pubchem_compound_name_search(spectrum_in: Spectrum,
     if spectrum_in is None:
         return None
     spectrum = spectrum_in.clone()
-    if _check_fully_annotated(spectrum):
+    # Only run this function if it does not yet have a useful annotation
+    if is_valid_inchi(spectrum.get("inchi")) or is_valid_smiles(spectrum.get("smiles")):
         return spectrum
```

Comment:
Maybe add a logger.info after those lines, saying that no valid annotation was found and that a pubchem search will be carried out?

Reply:
Good suggestion. I added it at the end of the function, which will only run if it did not find a good match. So we can actually log that pubchem lookup was run but not successfully.